bool quit
string up_down
string pressed_key
